gluten-core/src/main/scala/org/apache/gluten/softaffinity/SoftAffinityManager.scala

```diff
@@ -88,6 +90,13 @@ abstract class AffinityManager extends LogLevelUtil with Logging {
         }
       })
 
+  private var hashRing: Option[ConsistentHash[ExecutorNode]] = _
```

Comment:
Why we need an Option here? Since we must init the hashRing, why not directly defined here?

Reply:
Because the strategy is fundamentally scalable, and there may be other strategies in the future besides the consistentHash strategy, I think we should keep a scalable implementation.

Reply:
Okey. Then maybe move the `hashRing` to ConsistentHashSoftAffinityStrategy, which I think will be better expanded

Reply:
Thanks for you suggestion, only the manager is aware of the changes in the executor now, moving to the strategy requires that the strategy has the relevant interfaces for the executor's changes. the current soft affinity code is not very friendly in terms of extensibility, and we can refactor it in subsequent PRs to achieve better extensibility.